cineast-core/src/main/java/org/vitrivr/cineast/core/features/Superpixel.java

```diff
@@ -0,0 +1,256 @@
+import boofcv.factory.segmentation.FactorySegmentationAlg;
+import boofcv.gui.feature.VisualizeRegions;
+import boofcv.io.image.ConvertBufferedImage;
+import boofcv.struct.feature.ColorQueue_F32;
+import boofcv.struct.image.GrayF32;
+import boofcv.struct.image.GrayS32;
+import boofcv.struct.image.ImageBase;
+import boofcv.struct.image.ImageType;
+import boofcv.struct.image.Planar;
+import java.awt.AlphaComposite;
+import java.awt.Graphics2D;
+import java.awt.image.BufferedImage;
+import org.ddogleg.struct.FastQueue;
+import org.ddogleg.struct.GrowQueue_I32;
+import org.vitrivr.cineast.core.color.ColorConverter;
+import org.vitrivr.cineast.core.color.LabContainer;
+import org.vitrivr.cineast.core.color.RGBContainer;
+import org.vitrivr.cineast.core.data.segments.SegmentContainer;
+
+public class Superpixel {
+
+  // definition of superpixel algorithms
+  public static final int ALG_FH04 = 1; // Felzenszwalb-Huttenlocher
+  public static final int ALG_MS = 2; // MeanShift
+  public static final int ALG_SLIC = 3; // SLIC
+  public static final int ALG_WATERSHED = 4; // Watershed
+
+  // definition of image transformation
+  public static final int IMG_AVG = 1; // average frame
+  public static final int IMG_MED = 2; // median frame
+  public static final int IMG_REP = 3; // most representative frame
+
+  /**
+   * Code Inspiration Source: https://boofcv.org/index.php?title=Example_Superpixels Segments an
+   * image and calls visualization method
+   */
+  public static <T extends ImageBase<T>>
+  BufferedImage performSegmentation(ImageSuperpixels<T> alg, T color) {
+    GBlurImageOps.gaussian(color, color, 0.5, -1, null);
+    GrayS32 pixelToSegment = new GrayS32(color.width, color.height);
+    alg.segment(color, pixelToSegment);
+    return computation(pixelToSegment, color, alg.getTotalSuperpixels());
+  }
+
+  /**
+   * Code source: https://boofcv.org/index.php?title=Example_Superpixels
+   * <p>
+   * Don't need visualization, just computation of Buffered Image!
+   * <p>
+   */
+  public static <T extends ImageBase<T>>
+  BufferedImage computation(GrayS32 pixelToRegion, T color, int numSegments) {
+    // Computes the mean color inside each region
+    ImageType<T> type = color.getImageType();
+    ComputeRegionMeanColor<T> colorize = FactorySegmentationAlg.regionMeanColor(type);
+
+    FastQueue<float[]> segmentColor = new ColorQueue_F32(type.getNumBands());
+    segmentColor.resize(numSegments);
+
+    GrowQueue_I32 regionMemberCount = new GrowQueue_I32();
+    regionMemberCount.resize(numSegments);
+
+    ImageSegmentationOps.countRegionPixels(pixelToRegion, numSegments, regionMemberCount.data);
+    colorize.process(color, pixelToRegion, regionMemberCount, segmentColor);
+
+    // ATTENTION:
+    // When uncommenting the following line, the superpixels get colored according to the colors specified in ColorSuperpixel.java
+    // segmentColor = usePredefinedColorSet(segmentColor);
```

Comment:
If this is supposed to be an usable option, it should be built such that one does not need to uncomment code recompile the entire project in order to use it.